### ISSUE TITLE
[Proposal] Use swagger-codegen to generate wrapper api

### ISSUE THREAD

**Issue by edersonbrilhante** [DefectDojo/defectdojo_api#37]
I am using [swagger-codegen] (https://github.com/swagger-api/swagger-codegen) to generate a better mirror of defectdojo API. 
This way it is not needed to add manually new paths, actions, payload, etc  It is needed just to run swagger-codegen using the latest version of swagger file provided by defectdojo in ```/api/v2/doc/?format=openapi```

Here is my project running with this purpose https://github.com/William-Hill-Online/defectdojo-api-swagger

**Comment by edersonbrilhante** [comment]
Indeed, but I downloaded with my user logged, so I got details that was not avaiable not logged
```curl -X GET \
  '<endpoint>/api/v2/doc/?format=openapi' \
  -H 'Authorization: Token<token>' ```

**Comment by edersonbrilhante** [comment]
If the swagger is well configured in django-DefectDojo/dojo/api_v2/views.py I think it will not be a problem.

**Comment by edersonbrilhante** [comment]
Hi @valentijnscholten I fixed here https://github.com/DefectDojo/django-DefectDojo/pull/2488

**Comment by alles-klar** [comment]
I created a similar project a while ago: https://github.com/alles-klar/defectdojo-api-v2-client. Generally it works fine but there are some bugs in the defectdojo swagger documentation. So for a stable api client we need to write a lot of tests. @edersonbrilhante do you have an idea how to automatically generate tests?
